src/phpDocumentor/Reflection/Php/Factory/File.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+
+namespace phpDocumentor\Reflection\Php\Factory;
+
+use InvalidArgumentException;
+use phpDocumentor\Reflection\Php\File as FileElement;
+use phpDocumentor\Reflection\Php\NodesFactory;
+use phpDocumentor\Reflection\Php\ProjectFactoryStrategy;
+use phpDocumentor\Reflection\Php\StrategyContainer;
+use PhpParser\Lexer;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_ as ClassNode;
+use PhpParser\Node\Stmt\Function_ as FunctionNode;
+use PhpParser\Node\Stmt\Interface_ as InterfaceNode;
+use PhpParser\Node\Stmt\Namespace_ as NamespaceNode;
+use PhpParser\Node\Stmt\Trait_ as TraitNode;
+
+/**
+ * Strategy to create File element from the provided filename.
+ */
+final class File implements ProjectFactoryStrategy
+{
+    /**
+     * @var NodesFactory
+     */
+    private $nodesFactory;
+
+    /**
+     * Initializes the object
+     * @param NodesFactory $nodesFactory
+     */
+    public function __construct(NodesFactory $nodesFactory)
+    {
+        $this->nodesFactory = $nodesFactory;
+    }
+
+    /**
+     * Returns true when the strategy is able to handle the object.
+     *
+     * @param object $object object to check.
+     * @return boolean
+     */
+    public function matches($object)
+    {
+        return is_string($object) && file_exists($object);
+    }
+
+    /**
+     * Creates an File out of the given object.
+     * Since an object might contain other objects that need to be converted the $factory is passed so it can be
+     * used to create nested Elements.
+     *
+     * @param string $object object to convert to an File
```

Comment:
This line and param name confuses me; what do I provide? And do you think it could receive a Path object?